cookbooks/aws-parallelcluster-slurm/templates/default/slurm/login_nodes_daemon_service/login_nodes_on_termination_script.sh.erb

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+DEFAULT_USER=''
+OS=$(cat /etc/os-release | grep '^ID=' | cut -f2 -d'"')
+TERMINATION_MESSAGE=$1
+GRACETIME_PERIOD=$2
+
+if [[ $OS == "amzn" ]]; then
+    DEFAULT_USER='ec2-user'
+elif [[ $OS == "centos" ]]; then
+    DEFAULT_USER='centos'
+elif [[ $OS == "ubuntu" ]]; then
+    DEFAULT_USER='ubuntu'
+elif [[ $OS == "rhel" ]]; then
+    DEFAULT_USER='ec2-user'
+fi
+
+# Prevent further login/SSH attempts by updating ssh config
+echo "AllowUsers $DEFAULT_USER" >> /etc/ssh/sshd_config
+
+# Reload the ssh configuration
+systemctl reload sshd
+
+# Broadcast a message to all logged in users using the wall command
+wall "$TERMINATION_MESSAGE"
+
+# Sleep for the gracetime period
```

Comment:
*Minor*
It's a good practice to put comments on code section explaining what the code does.
Nevertheless, comments are expected to add informative value over the code they refer to; so if the code is self-explanatory, you can also avoid putting comments on every line of code.

In this specific case, for example, you can describe what the script is expected to do ion the introductory comment for the whole script and remove the comments on every pother line.

Reply:
Thanks for reminding. Done!